pkg/controller/cronjob/utils.go

```diff
@@ -249,15 +247,12 @@ func getJobFromTemplate2(cj *batchv1.CronJob, scheduledTime time.Time) (*batchv1
 	// We want job names for a given nominal start time to have a deterministic name to avoid the same job being created twice
 	name := getJobName(cj, scheduledTime)
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.CronJobsScheduledAnnotation) {
```

Comment:
I think this falls into the code that remains gated after being LockedToDefault so that the compatibility version continues to function properly.  @jpbetz could you double check me on this one and help me refine the understanding of the rule if it doesn't apply here?

Reply:
I was going to comment this also: https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/4330-compatibility-versions/README.md#summary.

I think we need to keep the feature gate code as kube-controller-manager is in scope for this.

Reply:
Currently we only have a nightly test - `compatibility-version-test`, dashboard here:
https://testgrid.k8s.io/sig-testing-kind#compatibility-version-test

I have created an issue against test-infra to create an triggerable version of the test:
https://github.com/kubernetes/test-infra/issues/33730